packages/compass-collection/src/components/collection-header-actions/collection-header-actions.spec.tsx

```diff
@@ -168,4 +204,184 @@ describe('CollectionHeaderActions [Component]', function () {
+      mockUseAssignment.returns({
+        assignment: {
+          assignmentData: {
+            variant: 'mockDataGeneratorControl',
+          },
+        },
+      });
+
+      await renderCollectionHeaderActions(
+        {
+          namespace: 'test.collection',
+          isReadonly: false,
+        },
+        {},
+        atlasConnectionInfo
+      );
+
+      expect(
+        screen.queryByTestId('collection-header-generate-mock-data-button')
+      ).to.not.exist;
+    });
+
+    it('should not show Mock Data Generator button when not in Atlas', async function () {
+      mockUseAssignment.returns({
+        assignment: {
+          assignmentData: {
+            variant: 'treatment',
+          },
+        },
+      });
+
+      await renderCollectionHeaderActions({
+        namespace: 'test.collection',
+        isReadonly: false,
+        // Don't pass atlasConnectionInfo, to simulate not being in Atlas
+      });
+
+      expect(
+        screen.queryByTestId('collection-header-generate-mock-data-button')
+      ).to.not.exist;
+    });
+
+    it('should not show Mock Data Generator button for readonly collections', async function () {
+      mockUseAssignment.returns({
+        assignment: {
+          assignmentData: {
+            variant: 'treatment',
+          },
+        },
+      });
+
+      await renderCollectionHeaderActions(
+        {
+          namespace: 'test.collection',
+          isReadonly: true,
+        },
+        {},
+        atlasConnectionInfo
+      );
+
+      expect(
+        screen.queryByTestId('collection-header-generate-mock-data-button')
+      ).to.not.exist;
+    });
+
+    it('should not show Mock Data Generator button for views (sourceName present)', async function () {
+      mockUseAssignment.returns({
+        assignment: {
+          assignmentData: {
+            variant: 'treatment',
+          },
+        },
+      });
+
+      await renderCollectionHeaderActions(
+        {
+          namespace: 'test.collection',
+          isReadonly: false,
+          sourceName: 'source-collection',
+        },
+        {},
+        atlasConnectionInfo
+      );
+
+      expect(
+        screen.queryByTestId('collection-header-generate-mock-data-button')
+      ).to.not.exist;
+    });
+
+    it('should show Mock Data Generator button when user is in treatment group and in Atlas', async function () {
+      mockUseAssignment.returns({
+        assignment: {
+          assignmentData: {
+            variant: 'mockDataGeneratorVariant',
+          },
+        },
+      });
+
+      await renderCollectionHeaderActions(
+        {
+          namespace: 'test.collection',
+          isReadonly: false,
+        },
+        {},
+        atlasConnectionInfo
+      );
+
+      expect(
+        screen.getByTestId('collection-header-generate-mock-data-button')
+      ).to.exist;
+      expect(screen.getByText('Generate Mock Data')).to.exist;
```

Comment:
get by text is already pretty good, get by role is even better, why do you duplicate this check with get by test id?